src/lambda/wizards/samInitWizard.ts

```diff
@@ -49,564 +33,258 @@ import {
-                    step: 1 + this.additionalSteps(),
-                    totalSteps: this.totalSteps + this.additionalSteps(),
-                },
-                buttons: [this.helpButton, vscode.QuickInputButtons.Back],
-                items: dependencyManagers.map(dependencyManager => ({
-                    label: dependencyManager,
-                })),
-            })
-
-            const choices = await picker.promptUser({
-                picker: quickPick,
-                onDidTriggerButton: (button, resolve, reject) => {
-                    if (button === vscode.QuickInputButtons.Back) {
-                        resolve(undefined)
-                    } else if (button === this.helpButton) {
-                        vscode.env.openExternal(vscode.Uri.parse(samInitDocUrl))
-                    }
-                },
-            })
-            const val = picker.verifySinglePickerOutput(choices)
-
-            return val ? (val.label as DependencyManager) : undefined
-        }
-    }
-
-    public async promptUserForArchitecture(
-        currRuntime: Runtime,
-        packageType: RuntimePackageType,
-        dependencyManager: DependencyManager
-    ): Promise<Architecture | undefined> {
-        // TODO: Remove Image Maven check if/when Hello World app supports multiarch Maven builds
-        if (!samArmLambdaRuntimes.has(currRuntime) || (dependencyManager === 'maven' && packageType === 'Image')) {
-            return 'x86_64'
-        }
-        this.stepsToAdd.promptUserForArchitecture = true
-        const quickPick = picker.createQuickPick<vscode.QuickPickItem>({
-            options: {
-                ignoreFocusOut: true,
-                title: localize('AWS.samcli.initWizard.architecture.prompt', 'Select an Architecture'),
-                step: 1 + this.additionalSteps(),
-                totalSteps: this.totalSteps + this.additionalSteps(),
-            },
-            buttons: [this.helpButton, vscode.QuickInputButtons.Back],
-            items: [{ label: 'x86_64' }, { label: 'arm64' }],
-        })
+function createRuntimePrompter(samCliVersion: string): QuickPickPrompter<RuntimeAndPackage> {
+    return createRuntimeQuickPick({
+        showImageRuntimes: semver.gte(samCliVersion, MINIMUM_SAM_CLI_VERSION_INCLUSIVE_FOR_IMAGE_SUPPORT),
+        buttons: createCommonButtons(samInitDocUrl),
+    })
+}
 
-        const choices = await picker.promptUser({
-            picker: quickPick,
-            onDidTriggerButton: (button, resolve, reject) => {
-                if (button === vscode.QuickInputButtons.Back) {
-                    resolve(undefined)
-                } else if (button === this.helpButton) {
-                    vscode.env.openExternal(vscode.Uri.parse(samInitDocUrl))
-                }
-            },
-        })
-        const val = picker.verifySinglePickerOutput(choices)
+function createTemplatePrompter(
```

Comment:
Nit: can we change this name from "Template" and "SAM Template" since they have a different meaning? OK with holding for now since this is established...just haven't looked at this code in a while and should've caught it the first time around.